### ISSUE TITLE
[Regression] there is no full response when complicated YG query sends

### ISSUE THREAD

**Issue by AnnaGrebnev** [thomsonreuters/CM-Well#529]
1. Send complicated YG query where it is required to bring data from the different graphs (the example of real use case will be provided by email)
Expected results: full response from all graphs appear
Actual results: partial response appears (there is no on part of the graph)

Some troubleshooting steps that I did;
1. If I remove one predicate from the query, it works as expected 
2. the predicate is Boolean vs the rest
3. Search by this predicate works as expected for both cases: full path and prefix

It impacts negatively our customer.

**Comment by hochgi** [comment]
To elaborate on the cause:
Before last release, we dealt with historic unmangled data by evaluating each filter as unmangled (which have string semantics when used as mangled field) as well as all type derivative fields.

e.g:
If we had a field `isActive.ns` which has domain of `boolean`s, the meta data stored to support it is:
an infoton under the ns id: `/meta/ns/id(ns)/isActive` which has a field `mang:['b']`.
The value `'b'` means that we should refer to values of that field as booleans.
This translate to a search in Elasticsearch with mangling prefixed fields. more specifically, ES sources for such infoton will contain:
`"fields" : {
  "${id(ns)}" : {
    "b$isActive" : [ true ]
  }
}`

if same field is used with other types, e.g. integers, the meta infoton of the field will have multiple values. e.g: 
```none
path = /meta/ns/${id(ns)}/isActive
fields = mang : [ 'b' , 'i' ]
```
if type `'s'` is used, which means a string, ES mappings have the bare field without an extra character and dollar sign prefixing it.

Now, as said above, old data didn't have strict types, and pre-migration environment had to deal with the dual mode. the hack was to always add the bare field name.

This worked in ES, since it was evaluated as nested should (per type).
It relates to current problem because the same `FieldFilter` class was used to model the filters to ES,
but also internally, when we used direct fetches from cassandra, we used those filters in memory with our own implementation to do the filters in expansion APIs (xg/yg/gqp).

Now, since `FieldFilter`s needs to be evaluated by translating prefixes to internal ns identifiers for example, we used the same evaluation process; which yielded the mangling field names, which while we used the dual mode hack, was OK, since bare field was part of the fields set.

Now, after we fixed the dual mode post migration, we also needed to patch the internal filtering to account for mangled fields.
The PR (#530) is a quick patch.
A more concise approach is needed here. since we now can count on mangling,
we can match against the correct types, and only check appropriate `FieldValue`s.
